website/static/js/addonSettings.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+
+var ConnectedProject = function(data) {
+    var self = this;
+    self.title = data.title;
+    self.id = data.id;
+    self.urls = data.urls;
+};
+
+var ExternalAccount = function(data) {
+    var self = this;
+    self.name = data.display_name;
+    self.id = data.id;
+    self.profileUrl = data.profile_url;
+
+    self.connectedNodes = ko.observableArray();
+
+    ko.utils.arrayMap(data.nodes, function(item) {
+        self.connectedNodes.push(new ConnectedProject(item));
+    });
+
+    self.deauthorizeNode = function(node) {
+        var url = node.urls.deauthorize;
+        var request = $.ajax({
+            url: url,
+            type: 'DELETE'
+        });
+        request.done(function(data) {
+            self.connectedNodes.remove(node);
+        });
+        request.fail(function(xhr, status, error) {
+            Raven.captureMessage('Error deauthorizing node: ' + node.id, {
+                url: url, status: status, error: error
+            });
+        });
+        return request;
+    };
+};
+
+var OAuthAddonSettingsViewModel = function(name, displayName) {
```

Comment:
JS tests should be added for this ViewModel (but they don't need to be added before this is merged).